cmd/commandfuncs.go

```diff
@@ -538,6 +539,135 @@ func cmdValidateConfig(fl Flags) (int, error) {
 	return caddy.ExitCodeSuccess, nil
 }
 
+type storVal struct {
+	StorageRaw json.RawMessage `json:"storage,omitempty" caddy:"namespace=caddy.storage inline_key=module"`
+}
+
+func DetermineStorage(configFile string, configAdapter string) (*storVal, error) {
+	cfg, _, err := LoadConfig(configFile, configAdapter)
+	if err != nil {
+		return nil, err
+	}
+
+	var tmpStruct storVal
+	err = json.Unmarshal(cfg, &tmpStruct)
+	if err != nil {
```

Comment:
This doesn't correctly handle the default case where no storage is defined in the config. My current thought is to check `json.Valid` first so that if `Unmarshal` returns `json.SyntaxError` it should be safe to assume it's the default case.

Reply:
That could probably work.